rebasehelper/tags.py

```diff
@@ -82,23 +84,43 @@ def _map_sections_to_parsed(cls, raw_content: SpecContent, parsed_content: SpecC
 
     @classmethod
     def _parse(cls, raw_content: SpecContent, parsed_content: SpecContent) -> Tuple[Tag]:
-        result = []
+        result: List[Tag] = []
         parsed_mapping = cls._map_sections_to_parsed(raw_content, parsed_content)
+        patch_source_indices = {tag: 0 for tag in ('source', 'patch')}
         # counts number of occurrences of one particular section
         for section_index, (section, section_content) in enumerate(raw_content.sections):
             section = section.lower()
             parsed_section_index = parsed_mapping[section_index]
             parsed: List[str] = [] if parsed_section_index == -1 else parsed_content.sections[parsed_section_index][1]
             if section.startswith('%package'):
-                result.extend(cls._parse_package_tags(section, section_content, parsed, section_index))
+                result.extend(cls._parse_package_tags(section, section_content, parsed, patch_source_indices,
+                                                      section_index))
             elif section.startswith('%sourcelist') or section.startswith('%patchlist'):
-                result.extend(cls._parse_list_tags(section, section_content, parsed,
-                                                   cast(Tuple[Tag], tuple(result)), section_index))
+                result.extend(cls._parse_list_tags(section, section_content, parsed ,patch_source_indices,
+                                                   section_index))
         return cast(Tuple[Tag], tuple(result))
 
+    @classmethod
+    def _sanitize_tag(cls, name: str, indices_dict: Dict[str, int]) -> Tuple[str, Optional[int]]:
```

Comment:
I don't particularly like this name, it's not descriptive enough, but unfortunately I can't think of anything better.